doc/source/serve/faq.rst

```diff
@@ -0,0 +1,247 @@
+to an endpoint through ``ServeHandle``, the request goes through the same code
+path as an HTTP request would: choosing backends through :ref:`traffic
+policies <serve-split-traffic>`, finding the next available replica, and
+batching requests together.
+
+When the request arrives in the model, you can access the data similarly to how
+you would with HTTP request. Here are some examples how ServeRequest mirrors Flask.Request:
+
+.. list-table::
+   :header-rows: 1
+
+   * - HTTP
+     - ServeHandle
+     - | Request
+       | (Flask.Request and ServeRequest)
+   * - ``requests.get(..., headers={...})``
+     - ``handle.options(http_headers={...})``
+     - ``request.headers``
+   * - ``requests.post(...)``
+     - ``handle.options(http_method="POST")``
+     - ``requests.method``
+   * - ``request.get(..., json={...})``
+     - ``handle.remote({...})``
+     - ``request.json``
+   * - ``request.get(..., form={...})``
+     - ``handle.remote({...})``
+     - ``request.form``
+   * - ``request.get(..., params={"a":"b"})``
+     - ``handle.remote(a="b")``
+     - ``request.args``
+   * - ``request.get(..., data="long string")``
+     - ``handle.remote("long string")``
+     - ``request.data``
+   * - ``N/A``
+     - ``handle.remote(python_object)``
+     - ``request.data``
+
+.. note::
+
+    You might have noticed that the last row of the table shows that ServeRequest supports
+    Python object pass through the handle. This is not possible in HTTP. If you
+    need to distinguish if the origin of the request is from Python or HTTP, you can do an ``isinstance``
+    check:
+
+    .. code-block:: python
+
+        import flask
+
+        if isinstance(request, flask.Request):
+            print("Request coming from web!")
+        elif isinstance(request, ServeRequest):
+            print("Request coming from Python!")
+
+.. note::
+
+    Once special case is when you pass a web request to a handle.
+
+    .. code-block:: python
+
+        handle.remote(flask_request)
+
+    In this case, Serve will `not` wrap it in ServeRequest. You can directly
+    process the request as a ``flask.Request``.
+
+How fast is ray serve? 
+----------------------
+We are continuously benchmark Ray Serve. We can confidently say Ray Serve's
+overhead is single digit millisecond, often times just 1-2 milliseconds. So
+that's the latency. What about throughput? Serve is horizontally scalable. This
+means you can just add more machines. If you double the machine, you can double
+your queries per second. On a single machine with few cores, Serve can achieve
+about 3-4k qps.
+
+How does Ray Serve make projects scalable without Flask?
+--------------------------------------------------------
+Flask is only used as a web request object for servable to consume the data.
+We actually use the fastest Python web server: uvicorn as our web server,
+alongside with the power of Python asyncio. To emphasize Flask is just the request
+object and data type. Not the web server.
```

Comment:
```suggestion
object and data type, not the web server.
```

Reply:
please link to uvicorn

Reply:
instead of saying to emphasize, put it in bold and state 

**Flask is ONLY the request object that we are using, Uvicorn, not flask, provides the webserver.**